### ISSUE TITLE
Evaluation on dev set

### ISSUE THREAD

**Issue by lethienhoa** [ChenRocks/fast_abs_rl#29]
Hi,
I noticed that on evaluation part of abstractor, you used ground-truth target to make prediction (not token of argmax prediction of the previous step). Even when we don't let the gradient backpropagate through this dev set, we should act like we don't really see this dataset, do we ?
I observed that the dev score is very close to training score by this bias (though training & dev are not that identical ?). This also leads to a quicker drop in learning rate & perhaps to a suboptimal ?

Thanks,
Hoa

**Comment by amobash2** [comment]
@lethienhoa I also noticed this behavior, I think many refer to this as "exposure bias" such as paper:
https://arxiv.org/pdf/1810.06667.pdf
(with reference to few other papers explaining this kind of bias).

When during training/evaluation model is trained by feeding ground-truth summary, while during test model should use word prediction (solve a kind of language model). The first paper suggested to use RL to deal with/reduce this exposure bias.

**Comment by ChenRocks** [comment]
At training abstractor we did validation using teacher forcing and compute loss mainly because decoding the full dev setis too slow.
However, at RL training and test phase of our full model we do not have access to the ground-truth so we generate word-by-word based on previous predictions.

Thanks for pointing out the exposure bias problem @amobash2 !
